### PR TITLE
bf(S3C-3970): Reduce scanning of unrelated events during metric calculation

### PR DESCRIPTION
**Pull request by tmacro**
also fixes a few off-by-one bugs reported in S3C-3997

### CODE REVIEW

File: warpscript/utapi/getMetricsAt.mc2 [hunk 1]
```diff
@@ -279,6 +308,7 @@
 
     $results // Leave results on the stack
 
+
```
Comment (jonathan-gramain) [inline warpscript/utapi/getMetricsAt.mc2]: unintended added whitespace?

**Review by jonathan-gramain — APPROVED** [review]
I did not review thoroughly all the logic, but did not see anything obviously wrong so approving.

**Review by rahulreddy — APPROVED** [review]

**Comment by rahulreddy** [comment]
@tmacro This PR should be targeted for stabilization/7.9.0

**Pull request merged**